proto/frontend/src/common.cpp

```diff
@@ -40,6 +40,14 @@ uint8_t clz(uint8_t b) {
   return (hb0 == 0) ? (4 + clz_table_hb[hb1]) : clz_table_hb[hb0];
 }
 
+uint8_t ctz(uint8_t b) {
+  static constexpr uint8_t ctz_table_hb[16] =
+      {4, 0, 1, 0, 2, 0, 1, 0, 3, 0, 1, 0, 2, 0, 1, 0};
+  uint8_t hb0 = b >> 4;
+  uint8_t hb1 = b & 0x0f;
```

Comment:
Nit: I would prefer more readable names like high_nibble and low_nibble, but this conforms to existing convention within this file.